### DISCUSSION TITLE
How to add a package to spack?

### DISCUSSION THREAD

**Discussion by jonas-eschle** [spack/spack#39173] [Q&A]
This is an insanely basic question, apologies if I was simply unlucky, but hours of googling didn't answer that to me: I've played around creating a "package.py", but how do I distribute this?
Making a PR and adding it to the repo?
Most likely. I didn't find any documentation, merge procedure, requirements,... anything.

Could you maybe point me to the docs where this is described?

**Comment by hhoeflin** [comment]
Please look here in the manual:

https://spack.readthedocs.io/en/latest/contribution_guide.html

**Reply by jonas-eschle** [reply]
Hm, I've seen that but I still don't quite understand: should we just add packages in the right folder? Just the package.py? Or anything else? As said, I didn't really find any information about this.

Is there any automation, i.e. in the style of conda-forge to update when a new version comes out? Any linting, testing that is centralized? I went through the "package guide" (as much as you can whithin a few hours) but it seems all a bit random to me.

**Reply by hhoeflin** [reply]
Yeah I am also not a package contributor, so I don't know. I once did a small update to a package, and that was a simple PR. Given that the package repo is just a flat structure of subdirectories named by the software with a package.py, I would think a PR to that effect is what is expected. 

There is also this https://spack-tutorial.readthedocs.io/en/latest/tutorial_packaging.html in case you had not seen it.

If in doubt, I would think making a best-effort PR and wait for the feedback (on things like linting) is the easiest.

**Reply by jonas-eschle** [reply]
Thanks, I've seen the link that describes (extensively) how to create a package, but not how to actually distribute it.

But okay, then I'll just do a PR
